contributed_definitions/nyaml/NXspm.yaml

```diff
@@ -0,0 +1,832 @@
+category: application
+doc: |
+  Scanning Probe Microscopy (SPM) is a branch of microscopy that utilizes a physical probe to scan the surface of
+  sample and image it at the atomic level.
+  
+  The application class NXspm is designed as a skeleton and contains common technical concepts
+  for specific SPM sub-techniques such as STM, STS, AFM etc.
+type: group
+NXspm(NXsensor_scan):
+  (NXentry):
+    definition:
+      doc: |
+        Name of the definition that is used for the application.
+      enumeration: [NXspm]
+    experiment_technique:
+      doc: |
+        The technique of the experiment like STM, STS, AFM, etc.
+      enumeration: [STM, STS, AFM]
```

Comment:
Is this enumeration exhaustive? What if somebody does EC-AFM for example? Maybe just get rid of the enum.

Reply:
That should raise an error, we do not have any such app def currently we have only three of them.